libtapi: fix build on macOS 10.5

catap opened the pull request:
#### Description

It also introduce a shortcut via clang-11-bootstrap for macOS before 10.12 which was the first macOS that can build it by its toolchain.

Thus, it is simplified version of https://github.com/macports/macports-ports/pull/15603

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->

macOS 10.5.8 9L34 i386
Xcode 3.1.4 9M2809

macOS 10.11.6 15G22010 x86_64
Xcode 7.3.1 7D1014

macOS 10.12.6 16G2136 x86_64
Xcode 9.2 9C40b

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

Gcenx commented on devel/libtapi/Portfile:
```diff
@@ -99,26 +114,21 @@ if {${os.platform} eq "darwin" && ${os.major} < 11} {
 
 # needed for libtapi to find some required files
-configure.args-append   -DCMAKE_CXX_FLAGS="-I ${worksrcpath}/src/llvm/projects/clang/include -I ${workpath}/build/projects/clang/include"
+configure.cxxflags-append \
```
Any reason why you've changed this line?

catap replied:
`configure.cxx_stdlib libc++` adds to `cxxflags` `-stdlib` flag, which should be preserved. Old way dismiss it.

Gcenx replied:
Oh didn't notice that

catap commented on devel/libtapi/Portfile:
```diff
@@ -85,12 +85,27 @@ if {[string match macports-clang-* ${configure.compiler}]} {
 
-if {${os.platform} eq "darwin" && ${os.major} < 11} {
+# on macOS before 10.12 use clang-11-bootstrap
+if {${os.platform} eq "darwin" && ${os.major} < 16 && ${build_arch} ni [list ppc ppc64]} {
```
@Gcenx your condition are true for any non ppc and ppc64 system :)

Gcenx replied:
Ah yes, I need to finish my morning coffees first....

Gcenx commented on devel/libtapi/Portfile:
```diff
@@ -5,23 +5,23 @@ PortGroup               cmake 1.1
 PortGroup               github 1.0
 
-github.setup            tpoechtrager apple-libtapi 664b84
+github.setup            tpoechtrager apple-libtapi b7b5bdbfda9e8062d405b48da3b811afad98ae76
```
Just a nitpick, I just find the shorter commit easier to read over the longer ones.

Doesn't really make any difference

mascguy reviewed (approved):
LGTM.

Kirill, are there any other PRs that should be merged first? I'm thinking of this one specifically: [PR 15950 - clang-11-bootstrap: backport support of using_if_exists](https://github.com/macports/macports-ports/pull/15950)

catap commented:
@mascguy I've just removed clear usage of clang-11-bootstrap to make everyone not so nervous :) and added `clang_dependency` PG that should breake cross-dependnecies which may appears after enforcing `libcxx`

Well, I feel that now it seems more safe to merge without any trigger :)
